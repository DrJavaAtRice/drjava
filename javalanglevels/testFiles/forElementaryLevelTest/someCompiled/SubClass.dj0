class SubClass extends SuperClass {
  int i;
  int j;
}