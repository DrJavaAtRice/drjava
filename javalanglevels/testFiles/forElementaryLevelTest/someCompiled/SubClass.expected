// Language Level Converter line number map: dj*->java. Entries: 4
//     1->3         2->4         3->5         4->6    
class SubClass extends SuperClass {
  private final int i;
  private final int j;

  /** This method is automatically generated by the Language Level Converter. */
  public SubClass(int super_, char super_0, int i, int j) {
    super(super_, super_0);
    this.i = i;
    this.j = j;
  }

  /** This method is automatically generated by the Language Level Converter. */
  public int i() {
    return i;
  }

  /** This method is automatically generated by the Language Level Converter. */
  public int j() {
    return j;
  }

  /** This method is automatically generated by the Language Level Converter. */
  public java.lang.String toString() {
    return getClass().getName() + "(" + 
        i() + ", " + 
        j() + ", " + 
        k() + ", " + 
        c() + 
        ")";
  }

  /** This method is automatically generated by the Language Level Converter. */
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    else if ((o == null) || (! o.getClass().equals(getClass()))) {
      return false;
    }
    else {
      SubClass cast = ((SubClass) o);
      return (i() == cast.i()) && 
          (j() == cast.j()) && 
          (k() == cast.k()) && 
          (c() == cast.c());
    }
  }

  /** This method is automatically generated by the Language Level Converter. */
  public int hashCode() {
    return getClass().hashCode() ^ 
        i() ^ 
        j() ^ 
        k() ^ 
        c();
  }
}
